src/Helper/DownloadLinkUriGenerator.php

```diff
@@ -22,7 +23,13 @@ public function __construct(UrlGeneratorInterface $urlGenerator, UriSigner $uriS
 
     public function generate(DownloadLink $link) : string
     {
-        $uri = $this->urlGenerator->generate('download', ['uri' => strtr(base64_encode($link->getUri()), self::URL_UNSAFE_CHARS, self::URL_UNSAFE_REPLACES), 'name' => $link->getFilename()], UrlGeneratorInterface::ABSOLUTE_URL);
+        $downloadUri = $link->getUri();
+
+        if ($link->getRelCanonical()) {
+            $downloadUri .= self::WRAP_REL_CANONICAL.$link->getRelCanonical();
```

Comment:
what does this do? is it ... appending an interpolated string to `$downloadUri` ?

Reply:
Have altered the approach. I wasn't happy that the decoded uri couldn't just be used to access the file directly. I have amended it to pass the the canonical uri as a query parameter: e.g. `http://www.example.com/file.pdf?canonicalUri=http://www.example.com/canonical`